### PR TITLE
[s]Implement softbans

### PR DESCRIPTION
**Pull request by ghost**
:cl: More Robust Than You
admin: Admins can now Softban, which allows players to observe, but not play in any way.
/:cl:

### CODE REVIEW

**Comment by kevinz000** [comment]
What is the point of that

**Comment by praisenarsie** [comment]
literally look in irc kevinz

File: code/__HELPERS/mobs.dm [hunk 1]
```diff
@@ -488,3 +488,8 @@ Proc for attack log creation, because really why not
 			warning("Invalid speech logging type detected. [logtype]. Defaulting to say")
 			log_say(logmessage)
 
+/proc/antag_banned(mob/M)
```
Comment (Jordie0608) [inline code/__HELPERS/mobs.dm]: These do not need to be procs, just type out the calls.

File: code/datums/spawners_menu.dm [hunk 2]
```diff
@@ -49,6 +49,9 @@
 				owner.forceMove(get_turf(MS))
 				. = TRUE
 		if("spawn")
+			if(is_softbanned(owner))
+				to_chat(owner, "<span class='warning'>You cannot do that!</span>")
```
Comment (Jordie0608) [inline code/datums/spawners_menu.dm]: This should say that it's because you're banned.

File: code/modules/admin/topic.dm [hunk 3]
```diff
@@ -865,6 +865,11 @@
 		else
 			dat += "<td width='20%'><a href='?src=[REF(src)];[HrefToken()];jobban3=alien candidate;jobban4=[REF(M)]'>Alien</a></td>"
 
+		//Softban (technically a job-ban)
+		if(is_softbanned(M))
+			dat += "<tr bgcolor='ffeeaa'><th colspan='10'><a href='?src=[REF(src)];[HrefToken()];jobban3=Softban;jobban4=[REF(M)]'><b><font color=red>Softbanned!</font></b></a></th></tr><tr align='center'>"
```
Comment (Jordie0608) [inline code/modules/admin/topic.dm]: There's already a button for this in the player panel, you don't need both.

File: code/modules/mob/dead/new_player/new_player.dm [hunk 4]
```diff
@@ -34,18 +34,26 @@
 
 /mob/dead/new_player/proc/new_player_panel()
 	var/output = "<center><p><a href='byond://?src=[REF(src)];show_preferences=1'>Setup Character</a></p>"
+	var/softbanned = is_softbanned(src)
```
Comment (Jordie0608) [inline code/modules/mob/dead/new_player/new_player.dm]: This doesn't need to be a var.

**Review by Jordie0608 — CHANGES_REQUESTED** [review]

**Comment by Jordie0608** [comment]
Ok and for those of us who weren't in irc, what is the reason for this?

**Comment by deathride58** [comment]
MRTY's got a perma, but he's also a contributor with a PR that's being testmerged often.

Due to his perma, he's unable to observe his testmerge (without ban evading)

**Comment by ShizCalev** [comment]
real fix:
don't get perma'd

**Comment by MoreRobustThanYou** [comment]
@ShizCalev true, but apparently permas can only be appealed once a year. writing an appeal for January 1st.

**Comment by MoreRobustThanYou** [comment]
@Jordie0608 review handled, PR squashed.

**Comment by kevinz000** [comment]
make it so you can softban yourself so i can force myself to admin instead of playing HoS for 10 hours straight and i'll give you support ayy

**Comment by lzimann** [comment]
The idea of softban is stupid. Plus, all you reallly need to do to not be permabanned is not being a person no one wants around, apparently that's too hard.

**Comment by Dorsisdwarf** [comment]
I thought people who were permabanned weren't allowed to contribute to the server codebase.

**Comment by Jordie0608** [comment]
There is no such rule.

**Pull request closed without merging**